hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleContextImpl.java

```diff
@@ -413,23 +414,26 @@ public <T> T dispatchPrecedingTransaction(
     public <T> T dispatchChildTransaction(
             @NonNull final TransactionBody txBody,
             @NonNull final Class<T> recordBuilderClass,
-            @NonNull final Predicate<Key> callback) {
+            @NonNull final Predicate<Key> callback,
+            @NonNull final AccountID syntheticPayer) {
         final var childRecordBuilder = recordListBuilder.addChild(configuration());
-        return doDispatchChildTransaction(txBody, childRecordBuilder, recordBuilderClass, callback);
+        return doDispatchChildTransaction(syntheticPayer, txBody, childRecordBuilder, recordBuilderClass, callback);
```

Comment:
A synthetic payer should only be applied if the transaction does not have a transaction ID, not in every case.

Reply:
Great point; added `dispatch*()` overloads that don't take an explicit payer and instead:
  1. Assert the `TransactionID#accountID` is set; and,
  2. Delegate to the explicit payer overload with that `accountID`.